lib/utils/prompt/context_reader_test.go

```diff
@@ -186,6 +186,70 @@ func TestContextReader_ReadPassword(t *testing.T) {
 	})
 }
 
+func TestNotifyExit_restoresTerminal(t *testing.T) {
+	oldStdin := Stdin()
+	t.Cleanup(func() { SetStdin(oldStdin) })
```

Comment:
This global worries me. While this test is not marked parallel, tests of multiple packages _are_ run in parallel, and IIRC global variables are shared between all packages under test

Reply:
Only parallel tests run in parallel, tests not marked with `t.Parallel` (and not subtests of those) are run in serial; it might be worth it to do a `t.SetEnv("MustRunSerialized", "")` to ensure that we never accidentally parallelize this.

Reply:
Tests of separate packages are separate binaries, they shouldn't interact with each other.

Reply:
@espadolini if you `go test ./package/a/ ./package/b/`, they will run in parallel even if not marked parallel. @codingllama is right though, they are separate binaries and won't interact, I was misremembering.